src/main/groovy/edu/oregonstate/mist/classsearchapi/dao/ClassSearchDAO.groovy

```diff
@@ -0,0 +1,203 @@
+package edu.oregonstate.mist.classsearchapi.dao
+
+import com.fasterxml.jackson.core.type.TypeReference
+import com.fasterxml.jackson.databind.ObjectMapper
+import edu.oregonstate.mist.api.jsonapi.ResourceObject
+import edu.oregonstate.mist.classsearchapi.core.Attributes
+import edu.oregonstate.mist.classsearchapi.core.Faculty
+import edu.oregonstate.mist.classsearchapi.core.MeetingTime
+import org.apache.http.HttpEntity
+import org.apache.http.client.HttpClient
+import org.apache.http.client.methods.CloseableHttpResponse
+import org.apache.http.util.EntityUtils
+
+class ClassSearchDAO {
+    private static final String STATUS_CLOSED = 'Closed'
+    private static final String STATUS_OPEN = 'Open'
+    private static final String STATUS_WAITLISTED = 'Waitlisted'
+
+    private UtilHttp utilHttp
+    private HttpClient httpClient
+    private ObjectMapper mapper = new ObjectMapper()
+
+    ClassSearchDAO(UtilHttp utilHttp, HttpClient httpClient) {
+        this.httpClient = httpClient
+        this.utilHttp = utilHttp
+    }
+
+    /**
+     * Performs class search and returns results in jsonapi format
+     *
+     * @param term
+     * @param subject
+     * @param courseNumber
+     * @param q
+     * @param pageNumber
+     * @param pageSize
+     * @return
+     */
+    public def getData(String term, String subject, String courseNumber, String q,
+                       Integer pageNumber, Integer pageSize) {
+        CloseableHttpResponse response
+        def data = []
+        def sourcePagination = [totalCount: 0, pageOffset: 0, pageMaxSize: 0]
+
+        try {
+            def query = getQueryMap(term, subject, courseNumber, q, pageNumber, pageSize)
+            response = utilHttp.sendGet(query, httpClient)
+
+            HttpEntity entity = response.getEntity()
+            def entityString = EntityUtils.toString(entity)
+
+            data = this.mapper.readValue(entityString,
+                    new TypeReference<List<HashMap>>() {
+                });
+
+            sourcePagination = getSourcePagination(response.getAllHeaders())
+            EntityUtils.consume(entity)
+        } finally {
+            response?.close()
+        }
+
+        [data: getFormattedData(data), sourcePagination: sourcePagination]
+    }
+
+    private static getSourcePagination(headers) {
+        def headerMap = [:]
+        headers.each {
+            headerMap[it.name] = it.value
+        }
+
+        [
+            totalCount: headerMap['X-hedtech-totalCount']?.toInteger(),
+            pageOffset: headerMap['X-hedtech-pageOffset']?.toInteger(),
+            pageMaxSize: headerMap['X-hedtech-pageMaxSize']?.toInteger()
+        ]
```

Comment:
I didn't have it there before, but it was such a pain to access the keys/values due to how the headers were returned. If it wasn't because of the features that the apache httpbuilder library provides and the metrics that dropwizards bake in I would switch back to the groovy class. It was far fewer lines of code and time debugging to use groovy.